backend/helpers/csv_reader.py

```diff
@@ -0,0 +1,43 @@
+from rest_framework.response import Response
+from rest_framework import status
+from books.models import Book
+from helpers.csv_format_checker import CSVFormatChecker
+from rest_framework.request import Request
+from helpers.csv_exceptions import ExtraHeadersException, MissingHeadersException
+import pandas as pd
+
+
+class CSVReader:
+
+    def __init__(self, csv_import_type: str) -> None:
+        self.csv_import_type = csv_import_type
+
+    def read_csv(self, request: Request):
+        response_data = {}
+        csv_format_checker = CSVFormatChecker(self.csv_import_type)
+        csv = request.FILES["file"]
+        try:
```

Comment:
I personally think this could be a little cleaner if from this "try" down to the last exception (before malformed_data = ...) was extracted into a separate function like "preprocess_csv" or something like that, but up to you

Reply:
Agreed. I tried doing this, but actually tricker than might first think because could be returning a `Response` or not returning anything. And also the dataframe is created within this code. So it will actually end up being somewhat hacky and less clean I discovered.